lib/round-tracker.js

```diff
@@ -146,14 +154,55 @@ export async function mapCurrentMeridianRoundToSparkRound ({
     )
   }
 
-  await pgClient.query(`
+  // TODO: replace "new Date()" with the time when the NextRound event was committed on the chain
```

Comment:
To my knowledge there's no reliable time on chain

Reply:
> To my knowledge there's no reliable time on chain

I know.

Stations are also not able to stop the work exactly at the round end because they don't know how long the next retrieval check is going to take. 

I consider the round deadline an approximate value, a hint to Stations.

An alternative we can consider:
- Whenever a Station wants to start the next retrieval check, it calls SPARK API to obtain the current round (and tasks for that round). If the round number is the same as before, then it continues in selecting samples for that same round. If the round number goes up, then it restarts the task selection algorithm.
- Downsides:
  - Stations will make more requests, going from "one per round" to "one per task".
  - Since the Station does not know how much time is left in the current round, it can still perform a retrieval check that will finish in the next round and thus won't be rewarded.

WDYT?

As I wrote my comment, it seems to me that the alternative I described is better. It's also less reliant on the central party - in the future, Stations can listen to RoundStarted events directly by querying the RPC API.

Reply:
I removed the round `deadline` in [550b263](https://github.com/filecoin-station/spark-api/pull/80/commits/550b263991278282aaba029cf25eb1f6fffdae0c).

We can add it back in the future if (and only if) needed.

Reply:
And [46712c5](https://github.com/filecoin-station/spark-api/pull/80/commits/46712c5db93f52cea8dbc75646e060dae5ab7a5a)

Reply:
And [cfa9afa](https://github.com/filecoin-station/spark-api/pull/80/commits/cfa9afa4a68b9c9ee41b6631dc4e121ac452b676) 🙈

Reply:
I agree with your reasoning, removing deadlines is a good move!